extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryFrameProcessor.java

```diff
@@ -313,32 +340,23 @@ public Operator.Signal push(RowsAndColumns rac)
       public void completed()
       {
         try {
-          // resultRowsAndCols has reference to frameRowsAndCols
-          // due to the chain of calls across the ops
-          // so we can clear after writing to output
-          flushAllRowsAndCols(resultRowAndCols);
-          frameRowsAndCols.clear();
-
+          flushAllRowsAndCols(resultRowAndCols, rowId);
         }
         catch (IOException e) {
           throw new RuntimeException(e);
         }
-        finally {
-          frameRowsAndCols.clear();
-          resultRowAndCols.clear();
-        }
       }
     });
   }
 
   /**
    * @param resultRowAndCols Flush the list of {@link RowsAndColumns} to a frame
+   * @param rowId An AtomicInteger representing the current row ID
    * @throws IOException
    */
-  private void flushAllRowsAndCols(ArrayList<RowsAndColumns> resultRowAndCols) throws IOException
+  private void flushAllRowsAndCols(ArrayList<RowsAndColumns> resultRowAndCols, AtomicInteger rowId) throws IOException
```

Comment:
`resultRowAndCols` and `rowId` are fields of the class; there shouldn't be a need to passed them to a member method

Reply:
Fair point. Have made change to this method, as well as other methods wherever we were passing rowId as method parameter.